### DISCUSSION TITLE
Request to Make sha256 Attribute Optional in http_archive for Renovate Integration

### DISCUSSION THREAD

**Discussion by hans-brgs** [renovatebot/renovate#23692] [Suggest an Idea]
### Tell us more.

Dear Renovate Team,

As a new user of Renovate, I recently embarked on integrating it with a Bazel project using the [rules_boost](https://github.com/nelhage/rules_boost) repository . During this process, I discovered that without the `sha256` attribute in the [Bazel Workspace file](https://github.com/nelhage/rules_boost/blob/33461a3c666af36d7a080e4fd6f554725df44132/README.md?plain=1#L12-L23) of the rules_boost repository, Renovate was unable to detect the dependency and thus did not send any pull requests for updates.

Upon reviewing the [Renovate Bazel documentation](https://docs.renovatebot.com/bazel/#legacy-workspace-files), it appears that `sha256` is a required attribute. However, the maintainer of the rules_boost repository informed me that `sha256` might not have always been required, possibly added in a recent Bazel refactor. Unfortunately, I lack further details on this. ([to see the discussion](https://github.com/nelhage/rules_boost/issues/475)) 

Considering this, would it be feasible to make` sha256` optional in Renovate's dependency detection? I believe this adjustment could simplify integration and reduce confusion for users like myself, though I'm aware my understanding of Renovate is still maturing.

Thank you for your incredible work and for considering this request. 

Best regards,

**Comment by HonkingGoose** [comment]
The [Bazel `http_archive` docs](https://bazel.build/rules/lib/repo/http#http_archive) says you really should use `sha256` or `integrity`. I've copy/pasted some relevant parts of the docs:

> `http_archive`
> Downloads a Bazel repository as a compressed archive file, decompresses it, and makes its targets available for binding.
>
> ## Attributes
> 
> | Name | Description |
> | --- | --- |
> | `integrity`| _String; optional_ Expected checksum in Subresource Integrity format of the file downloaded.  This must match the checksum of the file downloaded. _It is a security risk to omit the checksum as remote files can change._ At best omitting this field will make your build non-hermetic. It is optional to make development easier but either this attribute or `sha256` should be set before shipping. |
> | `sha256` | _String; optional_ The expected SHA-256 of the file downloaded. This must match the SHA-256 of the file downloaded. _It is a security risk to omit the SHA-256 as remote files can change._ At best omitting this field will make your build non-hermetic. It is optional to make development easier but either this attribute or `integrity` should be set before shipping. |

To me it sounds a bit silly to allow Renovate users to opt-in to less secure dependency management. But as always, I'll let the maintainers decide. 😉

**Reply by hans-brgs** [reply]
Thank you for your extremely quick reply!

**Reply by rarkins** [reply]
If the spec says it's optional then we can also make it so. PRs welcome

**Reply by cpsauer** [reply]
👍🏻 Bazel definitely nudges people to provide the (optional) hash, but enough people don't, and there's enough ongoing use of WORKSPACE files, that I think it's definitely worth being tolerant of inputs without sha256.

Thanks all for being great :)
